src/app/dataset-view/additional-components/dataset-settings-component/tabs/webhooks/create-edit-subscription-modal/create-edit-subscription-modal.component.ts

```diff
@@ -0,0 +1,112 @@
+})
+export class CreateEditSubscriptionModalComponent extends BaseComponent implements OnInit {
+    @Input({ required: true }) public datasetBasics: DatasetBasicsFragment;
+    @Input() public subscriptionData: MaybeNull<CreateWebhookSubscriptionSuccess>;
+
+    public dropdownList: SubscribedEventType[] = [];
+    public secret: MaybeUndefined<string>;
+    public readonly WebhookSubscriptionStatus: typeof WebhookSubscriptionStatus = WebhookSubscriptionStatus;
+    public readonly WebhookSubscriptionModalAction: typeof WebhookSubscriptionModalAction =
+        WebhookSubscriptionModalAction;
+    public readonly ErrorSets: typeof ErrorSets = ErrorSets;
+
+    public activeModal = inject(NgbActiveModal);
+    private fb = inject(FormBuilder);
+    private webhooksService = inject(WebhooksService);
+    private clipboard = inject(Clipboard);
+
+    public ngOnInit(): void {
+        this.webhooksService
+            .eventTypes()
+            .pipe(takeUntilDestroyed(this.destroyRef))
+            .subscribe((eventTypes: string[]) => {
+                this.dropdownList = eventTypes.map((eventType) => {
+                    return { name: this.eventTypesMapper(eventType), value: eventType };
+                });
+                if (this.subscriptionData) {
+                    this.createSubscriptionForm.patchValue({
+                        targetUrl: this.subscriptionData.input.targetUrl,
+                        eventTypes: this.subscriptionData.input.eventTypes,
+                        label: this.subscriptionData.input.label,
+                    });
+                    this.secret = this.subscriptionData.secret;
+                }
+            });
+    }
+
+    public get targetUrlControl(): AbstractControl {
+        return this.createSubscriptionForm.controls.targetUrl;
+    }
+
+    public get eventTypesControl(): AbstractControl {
+        return this.createSubscriptionForm.controls.eventTypes;
+    }
+
+    public createSubscriptionForm = this.fb.nonNullable.group({
+        targetUrl: this.fb.control("", [Validators.required, Validators.pattern(AppValues.URL_PATTERN_ONLY_HTTPS)]),
+        eventTypes: [[] as string[], Validators.required],
+        label: this.fb.control("", []),
+    });
+
+    public createWebhook(): void {
+        const result = this.createSubscriptionForm.value as WebhookSubscriptionInput;
+        this.activeModal.close({ action: WebhookSubscriptionModalAction.CREATE, payload: result });
+    }
+
+    public updateWebhook(): void {
+        const result = this.createSubscriptionForm.value as WebhookSubscriptionInput;
+        this.activeModal.close({ action: WebhookSubscriptionModalAction.UPDATE, payload: result });
+    }
+
+    public verifyNow(): void {
+        this.activeModal.close({ action: WebhookSubscriptionModalAction.VERIFY });
+    }
+
+    public copyToClipboard(event: MouseEvent, text: string): void {
+        this.clipboard.copy(text);
+        changeCopyIcon(event);
+    }
```

Comment:
I have checked, we have this logic copied 7 times in the project. Think of doing a directive or a helper component.